### PR TITLE
autotools: update make uninstall and add test

### LINKED ISSUE #954: CI: test that "make uninstall" works

### PR DESCRIPTION
**Pull request by j-ogas**
Closes #954.

### CODE REVIEW

File: examples/Makefile.am [hunk 1]
```diff
@@ -154,3 +154,7 @@ uninstall-local:
 	rm -f $(DESTDIR)$(examplesdir)/copy/symlink-to-fileA
 	rm -f $(DESTDIR)$(examplesdir)/copy/symlink-to-fileB-A
 	rm -f $(DESTDIR)$(examplesdir)/copy/symlink-to-fileB-B
+
+uninstall-hook:
+	rm -rf --one-file-system $(docdir)
```
Comment (reidpr) [inline examples/Makefile.am:159]: Would `rmdir` work for these? They should be empty right? `rm -rf` is so high powered I'd like to only use it if absolutely necessary.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -313,6 +313,22 @@ jobs:
...
+          /var/tmp/from-tarball/libexec
+          /var/tmp/from-tarball/share
+          /var/tmp/from-tarball/share/doc
+          /var/tmp/from-tarball/share/man
+          /var/tmp/from-tarball/share/man/man1
+          /var/tmp/from-tarball/share/man/man7
+          EOF
+          # Install again.
+          sudo make $ch_makej install
+          $ch_prefix/from-tarball/bin/ch-run --version
```
Comment (reidpr) [inline .github/workflows/main.yml:331]: What about doing this in a new step after the install isn't needed any more? Then it won't need to be reinstalled.

**Review by reidpr — CHANGES_REQUESTED** [review]
Thanks! Couple more suggestions.

File: test/Makefile.am [hunk 3]
```diff
@@ -78,6 +78,7 @@ install-data-hook:
 uninstall-hook:
 	rm -f $(DESTDIR)$(testdir)/fixtures/symlink-to-tmp
 	rmdir $(DESTDIR)$(testdir)/fixtures || true
+	rmdir $$(find $(pkglibexecdir) -type d | sort -r)
```
Comment (reidpr) [inline test/Makefile.am:81]: Nice!

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -333,6 +333,23 @@ jobs:
         run: |
           $ch_prefix/from-tarball/bin/ch-test all
 
+      - name: ensure make uninstall works
```
Comment (reidpr) [inline .github/workflows/main.yml:336]:
```suggestion
      - name: make uninstall
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -333,6 +333,23 @@ jobs:
...
+        if: ${{ matrix.builder == 'docker' && matrix.keep_sudo }}
+        run: |
+          cd $ch_prefix/charliecloud-*
```
Comment (reidpr) [inline .github/workflows/main.yml:339]:
```suggestion
```

**Review by reidpr — APPROVED** [review]

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -333,10 +333,9 @@ jobs:
         run: |
           $ch_prefix/from-tarball/bin/ch-test all
 
-      - name: ensure make uninstall works
+      - name: make uninstall
         if: ${{ matrix.builder == 'docker' && matrix.keep_sudo }}
         run: |
-          cd $ch_prefix/charliecloud-*
```
Comment (j-ogas) [inline .github/workflows/main.yml]: I had to add this to get the CI to pass; the default working directory is not /var/tmp/from-tarball`.

**Review by reidpr — CHANGES_REQUESTED** [review]
Please fix failing tests, then I'll merge.

**Review by reidpr — APPROVED** [review]

**Pull request merged**